### ISSUE TITLE
Old versions get additional new line added

### ISSUE THREAD

**Issue by jeffdgr8** [Kotlin/dokka#3965]
**Describe the bug**
Using the versioning plugin, older version HTML files receive an extra new line whenever they're processed with `:dokkaGenerate`. When a project replaces the older version HTML files every time they're reprocessed when API docs are regenerated, this will continuously grow the size of these older version files.

**Expected behaviour**
Older version files shouldn't receive an extra new line character, only necessary changes should be made to the files.

**To Reproduce**
Run `:dokkaGenerate` on a project with older version docs and compare the processed files with the original to see the additional new line character in every HTML file.

**Dokka configuration**
Configuration of dokka used to reproduce the bug

```Kotlin
dependencies {
    dokkaPlugin(libs.dokka.versioning)
}

tasks.dokkaGeneratePublicationHtml {
    dokka.pluginsConfiguration.versioning {
        olderVersionsDir = projectDir.resolve("docs/older")
        // ...
    }
}
```

**Installation**
- Operating system: macOS
- Build tool: Gradle v8.11.1
- Dokka version: 2.0.0 (also 1.9.20)

**Additional context**
Reproducible with [my library, Kotbase](https://github.com/jeffdgr8/kotbase/pull/32). Just run `:dokkaGenerate` and observe the Git changes. The only changes will be the additional new line characters to the older version HTML files. Every subsequent run will add an additional new line.

**Comment by adam-enko** [comment]
Thanks for the report, I can observe the same behaviour when I checkout Kotbase and run it locally.

@whyoleg I don't think this is related to DGP. When I downgrade the Dokka Engine version to 1.9.20 (in `dokka-convention.gradle.kts` add this line: `dokka.dokkaEngineVersion.set("1.9.20")`) I don't see the same behaviour, so I think this is something inside Dokka Engine.

**Comment by jeffdgr8** [comment]
Yes, as noted, I have experienced this behavior with both the v1 and v2 Gradle plugins. I just got around to reporting it now!

**Comment by jeffdgr8** [comment]
Fixed in #3974
